alibi/explainers/integrated_gradients.py

```diff
@@ -543,20 +547,85 @@ def _format_target(target: Union[None, int, list, np.ndarray],
 
     Returns
     -------
-    Formatted target as a list.
+    Formatted target as a np.array.
 
     """
     if target is not None:
         if isinstance(target, int):
-            target = [target for _ in range(nb_samples)]
-        elif isinstance(target, list) or isinstance(target, np.ndarray):
-            target = [t.astype(int) for t in target]
+            target = np.array([target for _ in range(nb_samples)])
+        elif isinstance(target, list):
+            target = np.array(target)
+        elif isinstance(target, np.ndarray):
+            pass
         else:
             raise NotImplementedError
 
     return target
 
 
+def _check_target(output_shape: Tuple,
+                  target: Optional[np.ndarray],
+                  nb_samples: int) -> None:
+    """
+    Parameters
+    ----------
+    output_shape
+        Output shape of the tensorflow model
+    target
+        Target formatted as np array target.
+    nb_samples
+        Number of samples in the batch.
+
+    Returns
+    -------
+    None
+
+    """
+
+    if target is not None:
+
+        if not (target.dtype == int):
+            raise ValueError("Targets must be integers")
+
+        if target.shape[0] != nb_samples:
+            raise ValueError(f"First dimension in target must be the same as nb of samples. "
+                             f"Found target 1st dimension: {target.shape[0]}; nb samples: {nb_samples}")
+
+        if len(target.shape) > 2:
+            raise ValueError("Target must be a rank-1 or a rank-2 tensor. If target is a rank-2 tensor, "
+                             "each column contains the index of the corresponding dimension "
+                             "in the model's output tensor.")
+
+        if len(output_shape) == 1:
+            out_rank, target_rank = 1, len(target.shape)
+            tmax, tmin = target.max(axis=0), target.min(axis=0)
+
+            if tmax > 1:
+                raise ValueError(f"Target value {tmax} out of range for output shape = 1 ")
+
+        elif len(output_shape) == 2:
+            out_rank, target_rank = 1, len(target.shape)
```

Comment:
I fee like `out_rank` should be `2` here but it seems you are using different conventions which makes it more confusing...

Reply:
I.e. are we ignoring batch dimension in some cases but not others?

Reply:
no, because out_rank doesn't include the dimension referring to the number of samples

Reply:
we are not ignoring it only in the case of the squash output (output_shape = 1)

Reply:
But `if len(output_shape) == 1` then `out_rank=1` so we are not ignoring it in this case, but are ignoring it in all other cases?

It's just confusing to read that `out_rank=1` in both `len(output_shape) == 1` and `len(output_shape) == 2`... Is there a better way to code this up or at least comment in the code?

Reply:
(I think supporting squash-output for 2-class classification is giving a lot of headache here and elsewhere...)

Reply:
well it gave me a headache yesterday when I was writing these warnings and getting errors for squash output